src/main/java/com/rcv/GuiConfigController.java

```diff
@@ -192,36 +200,56 @@ public LocalDate fromString(String string) {
     choiceOvervoteRule.getItems().addAll(Tabulator.OvervoteRule.values());
     choiceOvervoteRule.getItems().remove(Tabulator.OvervoteRule.RULE_UNKNOWN);
     choiceMultiSeatTransferRule.getItems().addAll(Tabulator.MultiSeatTransferRule.values());
-    choiceMultiSeatTransferRule.getItems()
+    choiceMultiSeatTransferRule
+        .getItems()
         .remove(Tabulator.MultiSeatTransferRule.TRANSFER_RULE_UNKNOWN);
 
-    // Restricts text fields to non-negative integers
-    textFieldMaxRankingsAllowed.textProperty().addListener((observable, oldValue, newValue) -> {
-      if (!newValue.matches("\\d*")) {
-        textFieldMaxRankingsAllowed.setText(oldValue);
-      }
-    });
-    textFieldMaxSkippedRanksAllowed.textProperty().addListener((observable, oldValue, newValue) -> {
-      if (!newValue.matches("\\d*")) {
-        textFieldMaxSkippedRanksAllowed.setText(oldValue);
-      }
-    });
-    textFieldNumberOfWinners.textProperty().addListener((observable, oldValue, newValue) -> {
-      if (!newValue.matches("\\d*")) {
-        textFieldNumberOfWinners.setText(oldValue);
-      }
-    });
-    textFieldDecimalPlacesForVoteArithmetic.textProperty()
-        .addListener((observable, oldValue, newValue) -> {
-          if (!newValue.matches("\\d*")) {
-            textFieldDecimalPlacesForVoteArithmetic.setText(oldValue);
-          }
-        });
-    textFieldMinimumVoteThreshold.textProperty().addListener((observable, oldValue, newValue) -> {
-      if (!newValue.matches("\\d*")) {
-        textFieldMinimumVoteThreshold.setText(oldValue);
-      }
-    });
+    // Restrict text fields to non-negative integers and set default values where applicable
+    textFieldMaxRankingsAllowed
+        .textProperty()
+        .addListener(
+            (observable, oldValue, newValue) -> {
```

Comment:
If you can access the field itself from the listener args (maybe via observable?), perhaps you can avoid duplicating this same logic for each field? The amount of repetition here is unfortunate.

Reply:
Wasn't possible via observable unfortunately, but I made a new inner listener class in the latest commit.